### ISSUE TITLE
Future Support of WebP format

### ISSUE THREAD

**Issue by EugineKKK** [bumptech/glide#571]
Will be very great if you will add a support for WebP format. Witch includes loading WebP images and playing the animations.. Like the Fresco library does.

To add support of loading images shouldn't be hard.. the challenge is to add animation playing support.

Tnx.

**Comment by Sheepment** [comment]
Yes please provide an easy way to display webP animations it will be very appraised!

**Comment by sjudd** [comment]
WebP images are supported via the Android framework, I believe it's 4.2+. WebP animations are not. As usual, pull requests here would be welcome.

**Comment by JimmyET** [comment]
想要使用webp格式，但glide不支持，自己写一套工具，发现很多坑，请求Glide的大神们快点支持webp

**Comment by start141** [comment]
哈哈，楼上的不说英语，大神怕是看不懂啊

**Comment by JimmyET** [comment]
translate.google.com  @sjudd

**Comment by TWiStErRob** [comment]
@JimmyET Google translate only produces readable text if the sentences are simple, they also need to be grammatically correct and even then it mostly works only between latin languages. According to it you said:
> Webp format you want to use, but does not support glide, write your own set of tools and found that many pit, Glide's request Great God who quickly support webp

@start141 replied:
> Haha, upstairs do not speak English, do not understand the big fear is God ah

I don't really understand how God comes into picture when talking about image formats. Different word order in grammar of English and Chinese also produces kind of Yodaspeak. All in all the translated text sounds more like an really artsy poem than tech talk.

Please make your best effort to write comments in English here (and any other forum where you only see English entries by others), otherwise you're likely to have problems communicating your views. Worst case you can ask a friend to help translate if you don't speak English. Thank you for understanding! (I'm not a native English speaker either.)

**Comment by start141** [comment]
@TWiStErRob Yes, you said is right.
I said '哈哈，楼上的不说英语，大神怕是看不懂啊', mean 'haha, do not speaking English, somebody may not understand what you say '. Sorry, it just replied for @JimmyET

**Comment by jjhesk** [comment]
+1 it looks like we really need to make it supporting [webp](https://developers.google.com/speed/webp/gallery2#contents_of_this_gallery) as it has amazing data size advantage for mobile data.

**Comment by tmxdyf** [comment]
@TWiStErRob 棒棒哒

**Comment by googolmo** [comment]
Can 4.x display WebP image from internet?

**Comment by TWiStErRob** [comment]
@googolmo  yes, Glide v3 and v4 both support WebP (as long as the underlying device can decode it)
For Android 4.x support see https://github.com/bumptech/glide/issues/571#issuecomment-131404236

**Comment by HeSuxiang** [comment]
two years passed,  this feature still a feature

**Comment by TWiStErRob** [comment]
It probably involves native code, or at least packaging an external webp library. A pull request is welcome for this to create an integration library to handle this.

**Comment by behelit** [comment]
5 years has passed and it's still not available?

**Comment by sjudd** [comment]
There are integration libraries for both decoding webp images and animated webp images. I've seen various open and closed source implementations, though unfortunately I haven't seen one that's suitable to use here directly.

For animated webp images in particular, the good news is that the AOSP framesequence library seems to offer a plausible way forward. I'm making progress on this front, although relatively slowly. I'll try and provide a more detailed update in the near future.

**Comment by Whoisurdady** [comment]
I‘m confused why the owner lead us to this issue.It‘s opened in 2015, and does not close now.Any way, does anybody here has an experience of using fresco to decode animated webp, works with glide? It really sucks to have 2 image library in one project.

**Comment by sjudd** [comment]
#2765 seems related.

**Comment by meikaiss** [comment]
@zjupure 
GlideWebpDecoder  library  is really useful.      Anybody else want load animated webp, can use GlideWebpDecode.  I have tried.

**Comment by mnjit20** [comment]
@zjupure which version of glide is used by GlideWebpDecoder ? is this decoder is on latest version?

**Comment by zjupure** [comment]
@mnjit20 GlideWebpDecoder is developed based on glide 4.2.0; Glide updates too fast and modifies some interface. GlideWebpDecoder needs to make some change to adapter the latest glide version.

**Comment by 10045125** [comment]
> I have written a glide integration webp library to support playing animated webp images, see [GlideWebpDecoder](https://github.com/zjupure/GlideWebpDecoder). This is based on libwebp project and some code from [Fresco](https://github.com/facebook/fresco) project. Transparent or lossless static-webp images below 4.2.1 will be support after certain tests soon. Feel free to try for supporting webps and improve the project, thanks.

Cpu, memory usage is higher than Fresco

**Comment by geofurb** [comment]
Is animated webp support still on the roadmap?

**Comment by eman1986** [comment]
I'm very interested in seeing this available

**Comment by out0fmemory** [comment]
Is animated webp support still on the roadmap?

**Comment by aju1996** [comment]
Is animated webp support still on the roadmap?

**Comment by saaib** [comment]
> > I have written a glide integration webp library to support playing animated webp images, see [GlideWebpDecoder](https://github.com/zjupure/GlideWebpDecoder). This is based on libwebp project and some code from [Fresco](https://github.com/facebook/fresco) project. Transparent or lossless static-webp images below 4.2.1 will be support after certain tests soon. Feel free to try for supporting webps and improve the project, thanks.
> 
> Cpu, memory usage is higher than Fresco

It would be nice to see a list of requirements from the developers for the proposed code to fulfill, otherwise seems like shooting in the dark.

+1 for supporting animated .webp files.

Thanks

**Comment by HassanImtiaz** [comment]
We are using Glide 3.7.0, now for loading webp images do we need to do anything additional or library loads webp images by-default? our app has minimum sdk version 5.0.

**Comment by elomonaco** [comment]
This would be very nice to see supported, my company use tenor's api and they like to toss in webp files once in a while which isn't rendering and throwing errors. Google themselves recommend the webp format, so some traction would be awesome.

**Comment by shenguojun** [comment]
Hope to add support for animated WebP

**Comment by AllanThisgaard** [comment]
Hello, 

Could this issue be awarded a higher priority? 🙏
The secure FOSS chat app, Signal (https://signal.org/), does not support animated WebP content, affecting millions of users worldwide - and the Signal team [directly quote](https://github.com/signalapp/Signal-Android/issues/8593#issuecomment-462148109) the lack of support in the Glide library, as the reason for it:

> We support what Glide supports. Right now, Glide has no built-in webp support and defaults to the system. I don't know if they really support animated webp at all. You can see this issue for reference: https://github.com/bumptech/glide/issues/571
 There's some libraries out there that add webp support to Glide that Glide has acknowledged, but IMO if they're not good enough for Glide to merge in, they're probably not good enough for us either. Image decoding is a sensitive area.  Either way, not much action for us to take. Animated webp files will start working when Glide adds support.
